pylint/testutils/pyreverse.py

```diff
@@ -110,6 +110,35 @@ def get_functional_test_files(
     return test_files
 
 
+def get_functional_test_modules(
+    root_directory: Path,
+) -> list[FunctionalPyreverseTestfile]:
+    """Get all functional test files from the given directory."""
+    test_files = []
+    for path in root_directory.rglob("__init__.py"):
```

Comment:
It would be nice if we could decouple the discovery logic from the presence of `__init__.py` files, as it is not strictly required in Python and `pylint` also supports [running on namespace packages](https://pylint.pycqa.org/en/latest/user_guide/usage/run.html#with-implicit-namespace-packages).  
Also it could create problems if somebody wants to use this new feature to create functional tests for package diagrams with a nested package structures, due to the `rglob`.

To be honest I don't have an immediate and simple (and intuitive to use) in mind. 
One way would be to use the convention that every folder directly below `tests/pyreverse/functional/modules` (or after renaming, `tests/pyreverse/functional/packages` is one test case.  
The downside is, that I believe we do not have this convention/restriction for functional tests of class diagrams, which can lead to confusion.

Reply:
I have changed the code so that every subdirectory is treated as an own testcase. I think this makes the most sense although it is different from the other functional tests